azkaban-exec-server/src/main/java/azkaban/execapp/ExecMetrics.java

```diff
@@ -29,21 +32,45 @@
 public class ExecMetrics {
   public static final String NUM_RUNNING_FLOWS_NAME = "EXEC-NumRunningFlows";
   public static final String NUM_QUEUED_FLOWS_NAME = "EXEC-NumQueuedFlows";
-  public static final String PROJECT_DIR_CACHE_HIT_RATIO_NAME = "EXEC-ProjectDirCacheHitRatio";
-  public static final String FLOW_SETUP_TIMER_NAME = "EXEC-flow-setup-timer";
+  public static final String PROJECT_DIR_CACHE_HIT_RATIO_NAME = "project-dir-cache-hit-ratio";
+  public static final String FLOW_SETUP_TIMER_NAME = "flow-setup-timer";
+  public static final String FLOW_KILLING_COUNTER_NAME = "flow-killing-counter";
+  public static final String FLOW_TIME_TO_KILL_HISTOGRAM_NAME = "flow-time-to-kill-histogram";
+  public static final String FLOW_KILLED_METER_NAME = "flow-killed-meter";
+  public static final String FLOW_SUCCESS_METER_NAME = "flow-success-meter";
+  public static final String JOB_FAIL_METER_NAME = "job-fail-meter";
+  public static final String JOB_SUCCESS_METER_NAME = "job-success-meter";
+  public static final String JOB_KILLED_METER_NAME = "job-killed-meter";
 
   private final MetricsManager metricsManager;
   private Timer flowSetupTimer;
   private final ProjectCacheHitRatio projectCacheHitRatio;
+  private Counter flowKillingCounter;
+  private Histogram flowTimeToKillHistogram;
+  private Meter flowKilledMeter;
+  private Meter flowSuccessMeter;
+  private Meter jobFailMeter;
+  private Meter jobSuccessMeter;
+  private Meter jobKilledMeter;
+  // TODO ypadron-in: add metrics to measure the time between flow submission and flow execution
+  // preparation/start after clock skew issues in execution times are resolved.
```

Comment:
I am under the impression what we want to lookout for is flow in Preparation state for many minutes, sometimes even hours.  Is that assumption right?

Clock skew should not hamper that I guess. Hence it will be helpful to know what clock skews are we referring here.

Reply:
Here is the reason for the clock issues: https://github.com/azkaban/azkaban/pull/2481/files#diff-e86078574720dfac4b80b618a6ccafe9L435-L439
I discussed this with @li-ygerchikov and we agreed on removing them until it's fixed. Essentially with that issue metrics using execution times will not be accurate (depending on how servers synchronize their time, the differences could be in the order of minutes) and therefore unreliable.